packages/iris-grid/src/IrisGridTableModelTemplate.ts

```
@@ -1893,17 +1895,59 @@ class IrisGridTableModelTemplate<
 
   async setValues(edits: readonly EditOperation[] = []): Promise<void> {
```

Why is setValues being changed at all?

---

The main issue was that there are certain paste operations that would attempt to set values in ranges that technically aren't editable yet, but are still valid paste operations. 

For example, if a user were to paste 30 rows into an input table where:
- Main table content: 20 rows
- Current pending area: 2 rows
- Total editable area: 22 rows

only the first 22 rows would be processed (filling the existing editable area), while the remaining 8 rows would be silently ignored because they fell outside the current pending row range. 

This behaviour didn't occur before as `isEditableRange` incorrectly returned true for these out-of-bounds rows. This allowed paste operations to proceed without restriction. This also ties into the issues with aggregations, as those extra rows were mistakenly included in calculations due to isEditableRange returning true when it shouldn't have.